tests/common/widgets/range-input/options.ts

```diff
@@ -0,0 +1,220 @@
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+        },
+        widgetParams: { attribute: 'price' },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(
+        document.querySelector('.ais-RangeInput')
+      ).toMatchNormalizedInlineSnapshot(
+        normalizeSnapshot,
+        `
+        <div
+          class="ais-RangeInput"
+        >
+          <form
+            class="ais-RangeInput-form"
+          >
+            <label
+              class="ais-RangeInput-label"
+            >
+              <input
+                class="ais-RangeInput-input ais-RangeInput-input--min"
+                max="1000"
+                min="1"
+                placeholder="1"
+                step="1"
+                type="number"
+              />
+            </label>
+            <span
+              class="ais-RangeInput-separator"
+            >
+              to
+            </span>
+            <label
+              class="ais-RangeInput-label"
+            >
+              <input
+                class="ais-RangeInput-input ais-RangeInput-input--max"
+                max="1000"
+                min="1"
+                placeholder="1000"
+                step="1"
+                type="number"
+              />
+            </label>
+            <button
+              class="ais-RangeInput-submit"
+              type="submit"
+            >
+              Go
+            </button>
+          </form>
+        </div>
+      `
+      );
+    });
+
+    test('renders with initial refinements', async () => {
+      const searchClient = createMockedSearchClient();
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          initialUiState: {
+            indexName: {
+              range: {
+                price: '100:200',
+              },
+            },
+          },
+        },
+        widgetParams: { attribute: 'price' },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      expect(document.querySelector('.ais-RangeInput-input--min')).toHaveValue(
+        100
+      );
+      expect(document.querySelector('.ais-RangeInput-input--max')).toHaveValue(
+        200
+      );
+    });
+
+    test('renders with precision', async () => {
+      const searchClient = createMockedSearchClient();
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+        },
+        widgetParams: { attribute: 'price', precision: 2 },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      ['min', 'max'].forEach((target) => {
+        expect(
+          document.querySelector(`.ais-RangeInput-input--${target}`)
+        ).toHaveAttribute('step', '0.01');
+      });
+    });
+
+    test('refines on submit', async () => {
```

Comment:
not sure if worth testing, but we could add a test that only refines min or max, not both, to check it doesn't accidentally refine 0 or something.

Reply:
yes that's fine, although I meant to do that in a separate test